pkg/sdn/api/v1/conversion.go

```diff
@@ -2,27 +2,75 @@ package v1
 
 import (
 	kapi "k8s.io/kubernetes/pkg/api"
+	conversion "k8s.io/kubernetes/pkg/conversion"
 
 	oapi "github.com/openshift/origin/pkg/api"
-	"github.com/openshift/origin/pkg/sdn/api"
+	newer "github.com/openshift/origin/pkg/sdn/api"
+)
+
+const (
+	KnownInvalidVNID = ^uint(0)
```

Comment:
In versions v1beta3/v1, VNID was assigned by out of band sdn plugin and NetNamespace REST interface persisted the object. With the new api, VNID may or may not be assigned.

In the new api, VNID is a uint pointer and we know whether it is assigned or not but in v1beta3/v1,VNID is of type uint and it defaults to zero which is a valid id and there is no way to figure out if this is actually assigned by user or not. VNID can range from 0 to 2^24-1 (both inclusive). I've added 'KnownInvalidVNID' for backward compatibility so that any conversion from newer api --> v1/v1beta3 --> newer api will preserve the original intent.